### DISCUSSION TITLE
Ideal  image size for object detection

### DISCUSSION THREAD

**Discussion by TheCod3rMonk** [robmarkcole/HASS-Deepstack#241] [Q&A]
Hello,

I have been using the deepstack object detection integration in Home assistant to periodically scan images from 7 cameras in 5s intervals. All of these are 4K cameras and I am currently passing the full resolution 3840X2160 image from the main stream for processing. Apart from lower processing times in general, I was wondering if there are any other pros/cons on using lower resolution images for processing? For example, would it hurt the accuracy for object detection if I use a lower resolution 640X480 image vs. an image with 3840X2160 resolution? 
Much appreciate the help!

**Comment by robmarkcole** [comment]
The images are actually downsized by deepstack, and the size depends on the setting selected, see https://github.com/johnolafenwa/DeepStack/blob/dev/intelligencelayer/shared/shared.py

Assuming you are on high the image is downsized to 640x640

**Reply by TheCod3rMonk** [reply]
Thank you Dr. Robin. From shared.py, it seems like the downsized image resolution can be maximum at 640X640 or minimum at 256X256.  Just for my understanding, what would be the downsized image resolution if scanning from a 640X480 image?
I apologize if this is a silly question but just reiterating what I asked earlier, would scanning a 256x256 image be less accurate in detecting objects than a 640X640 image? I guess this is a question relating to the theory of image processing in general.
Thank you again for all your time.

**Reply by robmarkcole** [reply]
Not a silly question at all :-) The resizing function is at https://github.com/johnolafenwa/DeepStack/blob/d1712ea558025a72af16fdfc01141dedd4f7dbfc/intelligencelayer/shared/utils/datasets.py#L863 and uses the expand function from https://pillow.readthedocs.io/en/stable/reference/ImageOps.html#PIL.ImageOps.expand

So 640X480 would become 640X640 where the new 120 pixels in width are just zeros
